### PR TITLE
Enhance the editor to accept an already existing experience file or create a new experience

### PR DESCRIPTION
**Pull request by Ysabz**
### Summary 
This PR enhances the editor page to start up for a new experience or to load an already existing experience for modification. This PR also enhances the UI look and fixes some of the UI bugs.
### Details
- added abstraction to how the overlay recieves a data ( experience or apparatus json for a new experience) 
- fixes some of the UI bugs that caused UI buttons color change after clicking certain buttons 
### How to test?
- run the project. select "create a new experience) -> select an apparatus -> create a sequence of actions ( delete , add, reorder)  -> click on save button in navbar -> click on return button in navbar 
- select "load an existing experience) -> in the loaded editor page you should see the proper apparatus loading and the action sequencce box pre populated with the actions you saved in the previous step
### Checks 
- [ ] Tested Changes
- [ ] Stakeholder Approval

### CODE REVIEW

File: pages/experience.tsx [hunk 1]
```diff
@@ -30,29 +24,56 @@ const LoadingView = (): JSX.Element => (
...
+        experience.apparatusMetadata = apparatusJson.Metadata;
+        setExperienceData(experience);
+        setLoading(false);
+      });
+    } else {
+      getExperienceFromCloud(userId, dataId).then((experienceJson) => {
+        experience.initializationData.actionList = experienceJson.actionList;
+        getApparatusFromCloud(experienceJson.apparatusId).then(
+          (apparatusJson) => {
+            experience.apparatusId = apparatusJson.Id.Identifier;
```
Comment (mckenzietsean) [inline pages/experience.tsx:51]: Trying to load an experience crashes and displays "TypeError: apparatusJson.Id is undefined."
Also 2 code smells that need to be fixed.

**Review by mckenzietsean — CHANGES_REQUESTED** [review]

File: pages/experience.tsx (hunk 1 above)
Comment (Ysabz) [reply pages/experience.tsx:51]: It was caused by merging main into this branch which affected the experience json file fields names. Fixed now

Comment (mckenzietsean) [reply pages/experience.tsx:51]: Can confirm that I can now load apparatuses without it crashing.

**Comment by JustinLohKingWei** [comment]
Tested changes according to the provided instructions. All functions are working as intended.

File: pages/experience.tsx [hunk 2]
```diff
@@ -30,29 +24,56 @@ const LoadingView = (): JSX.Element => (
...
   const [loading, setLoading] = useState(true);
-  const [jsonFile, setJsonFile] = useState([]);
+  const [experienceData, setExperienceData] = useState<ExperienceData>();
+  // either apparatusID is provided or experience id but not both
+
+  const experience: ExperienceData = {
+    apparatusMetadata: { Paths: [], Data: [] },
+    apparatusId: "",
+    initializationData: { actionList: [] },
+  };
```
Comment (shushengfuqin) [inline pages/experience.tsx:37]: is it possible to annotate, for example userId the type of it.

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+          </NavButton>
+          <NavButton key="save" onClick={save}>
+            Save
+          </NavButton>
+        </Box>
+      </Toolbar>
+    </Container>
+  </AppBar>
+);
+export default Navbar;
```
Comment (shushengfuqin) [inline src/components/Navbar.tsx:42]: same thing with the annotation for the save and toggle if possible

File: src/components/webgl/WebglBox.tsx [hunk 4]
```diff
@@ -6,37 +6,35 @@ import { unityContext, load } from "../../util/unityContextActions";
...
+function WebglBox({ userId, experienceData }): JSX.Element {
+  useEffect(() => {
+    unityContext.on("loaded", () => {
+      // For some reason the unityContext.send("Container", "LoadApparatus", arg) in load() cannot be called at this point
+      // Having a timeout bypasses this
+      setTimeout(() => {
+        load(experienceData.apparatusId);
+      }, 100);
+    });
+  }, []);
```
Comment (shushengfuqin) [inline src/components/webgl/WebglBox.tsx:23]: once again annotation

**Review by shushengfuqin — COMMENTED** [review]
Please put the annotation.

**Comment by shushengfuqin** [comment]
Also for the 2 code smell, maybe take a look if they are left because it has to be there or will be solved later?

**Review by tashfia98 — COMMENTED** [review]
When "Create New Experience" button is clicked, Will it override the "testexp1" json already existing or create a new json with a new name?

**Review by mckenzietsean — APPROVED** [review]

File: pages/experience.tsx (hunk 2 above)
Comment (Ysabz) [reply pages/experience.tsx:37]: from annotation you mean defining the type of each variable?

File: src/components/Navbar.tsx (hunk 3 above)
Comment (Ysabz) [reply src/components/Navbar.tsx:42]: they are callback functions so I'm not sure if annotation fits

**Comment by Ysabz** [comment]
> 

Yes for now the exp file is hardocded so it always access and writes to the same file in the  cloud. In the next PRs the hardcoded name should be replaced with the name provided by the user ( experience name) so that read and write is done from the correct file

**Comment by Ysabz** [comment]
@JustinLohKingWei, for this PR the experience file name is hardcoded so it's always reading/writing from the same file. But in the next PRs this should be fixed by prompting the user for a name when creating a new experience and use that name for the file. same thing for overwrting the experience file, the original name of the experience file that was openend in the editor should be used

**Pull request merged**